tests/test_rank_models.py

```
@@ -0,0 +1,330 @@
+
+class TestRankChain:
+    def test_high_intelligence_free_fast_wins(self):
+        bench = _make_bench()
+        cats = _make_cats()
+        tier = rank_models.TierConfig(w=3, I_floor=0.55, n_out_tokens=4000, purpose="coding")
+        telemetry = {
+            ("nvidia", "gemini-3-flash"): _FakeStat(tps=50.0, ttft=400.0),
+            ("groq", "llama-small"): _FakeStat(tps=250.0, ttft=200.0),
+        }
+        chain = _make_chain()
+        new_chain, reasons = rank_models.rank_chain(chain, telemetry, {}, bench, cats, tier)
```

Ruff (0.15.18)

[warning] Unpacked variable `reasons` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)